### PR TITLE
fix(components): [select] backspace delete disabled option

### LINKED ISSUE #11991: [Component] [select] el-select mutiple filterable  could use `backspace` key to delete disabled options
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.47`
- Element Plus Version: `2.2.36`
- Browser / OS: `Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36`
- Build Tool: `Vite`

### Related Component

- `el-select`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZGl2IGNsYXNzPVwibS00XCI+XG4gICAgPHA+ZGVmYXVsdDwvcD5cbiAgICA8ZWwtc2VsZWN0XG4gICAgICB2LW1vZGVsPVwidmFsdWUxXCJcbiAgICAgIG11bHRpcGxlXG4gICAgICBwbGFjZWhvbGRlcj1cIlNlbGVjdFwiXG4gICAgICBmaWx0ZXJhYmxlXG4gICAgICBzdHlsZT1cIndpZHRoOiAyNDBweFwiXG4gICAgPlxuICAgICAgPGVsLW9wdGlvblxuICAgICAgICA6ZGlzYWJsZWQ9XCJpdGVtLmRpc2FibGVcIlxuICAgICAgICB2LWZvcj1cIml0ZW0gaW4gb3B0aW9uc1wiXG4gICAgICAgIDprZXk9XCJpdGVtLnZhbHVlXCJcbiAgICAgICAgOmxhYmVsPVwiaXRlbS5sYWJlbFwiXG4gICAgICAgIDp2YWx1ZT1cIml0ZW0udmFsdWVcIlxuICAgICAgLz5cbiAgICA8L2VsLXNlbGVjdD5cbiAgPC9kaXY+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0IGxhbmc9XCJ0c1wiIHNldHVwPlxuaW1wb3J0IHsgcmVmIH0gZnJvbSBcInZ1ZVwiO1xuXG5jb25zdCB2YWx1ZTEgPSByZWYoW1wiT3B0aW9uMVwiXSk7XG5cbmNvbnN0IG9wdGlvbnMgPSBbXG4gIHtcbiAgICB2YWx1ZTogXCJPcHRpb24xXCIsXG4gICAgbGFiZWw6IFwiT3B0aW9uMVwiLFxuXG4gICAgZGlzYWJsZTogdHJ1ZSxcbiAgfSxcbiAge1xuICAgIHZhbHVlOiBcIk9wdGlvbjJcIixcbiAgICBsYWJlbDogXCJPcHRpb24yXCIsXG4gICAgZGlzYWJsZTogZmFsc2UsXG4gIH0sXG4gIHtcbiAgICB2YWx1ZTogXCJPcHRpb24zXCIsXG4gICAgbGFiZWw6IFwiT3B0aW9uM1wiLFxuICAgIGRpc2FibGU6IGZhbHNlLFxuICB9LFxuICB7XG4gICAgdmFsdWU6IFwiT3B0aW9uNFwiLFxuICAgIGxhYmVsOiBcIk9wdGlvbjRcIixcbiAgICBkaXNhYmxlOiBmYWxzZSxcbiAgfSxcbiAge1xuICAgIHZhbHVlOiBcIk9wdGlvbjVcIixcbiAgICBsYWJlbDogXCJPcHRpb241XCIsXG4gICAgZGlzYWJsZTogZmFsc2UsXG4gIH0sXG5dO1xuPC9zY3JpcHQ+XG4iLCJpbXBvcnRfbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsIl9vIjp7fX0=)

### Steps to reproduce

press backspace key on the el-select component(it has the disabled option on v-model) 

### What is Expected?

the default disabled option should not be removed

### What is actually happening?

the disabled option is removed

### Additional comments

maybe this is the source code here defines how we press backspace  key event
https://github.com/element-plus/element-plus/blob/dev/packages/components/select/src/useSelect.ts#L606-L617

### PR DESCRIPTION
**Pull request by MrWeilian**
closed: #11991 

Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by MrWeilian** [comment]
[playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xuaW1wb3J0IHsgdmVyc2lvbiBhcyBFcFZlcnNpb24gfSBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBFbGVtZW50UGx1cyB9IGZyb20gJ0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlJ1xuXG5jb25zdCB2YWx1ZTEgPSByZWYoW1wiT3B0aW9uMlwiLCBcIk9wdGlvbjFcIl0pO1xuY29uc3Qgb3B0aW9ucyA9IFtcbiAge1xuICAgIHZhbHVlOiBcIk9wdGlvbjFcIixcbiAgICBsYWJlbDogXCJPcHRpb24xXCIsXG5cbiAgICBkaXNhYmxlOiB0cnVlLFxuICB9LFxuICB7XG4gICAgdmFsdWU6IFwiT3B0aW9uMlwiLFxuICAgIGxhYmVsOiBcIk9wdGlvbjJcIixcbiAgICBkaXNhYmxlOiBmYWxzZSxcbiAgfSxcbiAge1xuICAgIHZhbHVlOiBcIk9wdGlvbjNcIixcbiAgICBsYWJlbDogXCJPcHRpb24zXCIsXG4gICAgZGlzYWJsZTogZmFsc2UsXG4gIH0sXG4gIHtcbiAgICB2YWx1ZTogXCJPcHRpb240XCIsXG4gICAgbGFiZWw6IFwiT3B0aW9uNFwiLFxuICAgIGRpc2FibGU6IGZhbHNlLFxuICB9LFxuICB7XG4gICAgdmFsdWU6IFwiT3B0aW9uNVwiLFxuICAgIGxhYmVsOiBcIk9wdGlvbjVcIixcbiAgICBkaXNhYmxlOiBmYWxzZSxcbiAgfSxcbl07XG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICAgPGVsLXNlbGVjdFxuICAgIHYtbW9kZWw9XCJ2YWx1ZTFcIlxuICAgIG11bHRpcGxlXG4gICAgcGxhY2Vob2xkZXI9XCJTZWxlY3RcIlxuICAgIGZpbHRlcmFibGVcbiAgICBzdHlsZT1cIndpZHRoOiAyNDBweFwiXG4gID5cbiAgICA8ZWwtb3B0aW9uXG4gICAgICA6ZGlzYWJsZWQ9XCJpdGVtLmRpc2FibGVcIlxuICAgICAgdi1mb3I9XCJpdGVtIGluIG9wdGlvbnNcIlxuICAgICAgOmtleT1cIml0ZW0udmFsdWVcIlxuICAgICAgOmxhYmVsPVwiaXRlbS5sYWJlbFwiXG4gICAgICA6dmFsdWU9XCJpdGVtLnZhbHVlXCJcbiAgICAvPlxuICA8L2VsLXNlbGVjdD5cblxuICA8cD5cbiAgICA8ZWwtaWNvbiBjb2xvcj1cInZhcigtLWVsLWNvbG9yLXByaW1hcnkpXCI+PEVsZW1lbnRQbHVzIC8+PC9lbC1pY29uPlxuICAgIEVsZW1lbnQgUGx1cyB7eyBFcFZlcnNpb24gfX0gKyBWdWUge3sgdnVlVmVyc2lvbiB9fVxuICA8L3A+XG48L3RlbXBsYXRlPlxuIiwiUGxheWdyb3VuZE1haW4udnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCBBcHAgZnJvbSAnLi9BcHAudnVlJ1xuaW1wb3J0IHsgc2V0dXBFbGVtZW50UGx1cyB9IGZyb20gJy4vZWxlbWVudC1wbHVzLmpzJ1xuc2V0dXBFbGVtZW50UGx1cygpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8QXBwIC8+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0X21hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwiZWxlbWVudC1wbHVzXCI6IFwiaHR0cHM6Ly9wcmV2aWV3LTExOTk1LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguZnVsbC5taW4ubWpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXMvXCI6IFwidW5zdXBwb3J0ZWRcIlxuICB9XG59IiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwidnVlXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AdnVlL3J1bnRpbWUtZG9tQGxhdGVzdC9kaXN0L3J1bnRpbWUtZG9tLmVzbS1icm93c2VyLmpzXCIsXG4gICAgXCJAdnVlL3NoYXJlZFwiOiBcImh0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vQHZ1ZS9zaGFyZWRAbGF0ZXN0L2Rpc3Qvc2hhcmVkLmVzbS1idW5kbGVyLmpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXNcIjogXCJodHRwczovL3ByZXZpZXctMTE5OTUtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5mdWxsLm1pbi5tanNcIixcbiAgICBcImVsZW1lbnQtcGx1cy9cIjogXCJ1bnN1cHBvcnRlZFwiLFxuICAgIFwiQGVsZW1lbnQtcGx1cy9pY29ucy12dWVcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlQDIvZGlzdC9pbmRleC5taW4uanNcIlxuICB9LFxuICBcInNjb3Blc1wiOiB7fVxufSIsImVsZW1lbnQtcGx1cy5qcyI6ImltcG9ydCB7IGdldEN1cnJlbnRJbnN0YW5jZSB9IGZyb20gJ3Z1ZSdcbmltcG9ydCBFbGVtZW50UGx1cyBmcm9tICdlbGVtZW50LXBsdXMnXG5cbmxldCBpbnN0YWxsZWQgPSBmYWxzZVxuYXdhaXQgbG9hZFN0eWxlKClcblxuZXhwb3J0IGZ1bmN0aW9uIHNldHVwRWxlbWVudFBsdXMoKSB7XG4gIGlmIChpbnN0YWxsZWQpIHJldHVyblxuICBjb25zdCBpbnN0YW5jZSA9IGdldEN1cnJlbnRJbnN0YW5jZSgpXG4gIGluc3RhbmNlLmFwcENvbnRleHQuYXBwLnVzZShFbGVtZW50UGx1cylcbiAgaW5zdGFsbGVkID0gdHJ1ZVxufVxuXG5leHBvcnQgZnVuY3Rpb24gbG9hZFN0eWxlKCkge1xuICByZXR1cm4gbmV3IFByb21pc2UoKHJlc29sdmUsIHJlamVjdCkgPT4ge1xuICAgIGNvbnN0IGxpbmsgPSBkb2N1bWVudC5jcmVhdGVFbGVtZW50KCdsaW5rJylcbiAgICBsaW5rLnJlbCA9ICdzdHlsZXNoZWV0J1xuICAgIGxpbmsuaHJlZiA9ICdodHRwczovL3ByZXZpZXctMTE5OTUtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5jc3MnXG4gICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdsb2FkJywgcmVzb2x2ZSlcbiAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2Vycm9yJywgcmVqZWN0KVxuICAgIGRvY3VtZW50LmJvZHkuYXBwZW5kKGxpbmspXG4gIH0pXG59IiwiX28iOnsic2hvd0hpZGRlbiI6dHJ1ZSwic3R5bGVTb3VyY2UiOiJodHRwczovL3ByZXZpZXctMTE5OTUtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5jc3MifX0=)

File: packages/components/select/src/useSelect.ts [hunk 1]
```diff
@@ -603,10 +604,23 @@ export const useSelect = (props, states: States, ctx) => {
...
+    let lastNotDisabledIndex = -1
+    for (let i = value.length - 1; i >= 0; i--) {
+      if (!states.disabledOptions.has(value[i])) {
+        lastNotDisabledIndex = i
+        break
+      }
+    }
+    return lastNotDisabledIndex
+  }
+
```
Comment (MrWeilian) [inline packages/components/select/src/useSelect.ts:617]: 从最右边开始，找到第一个非 `disabled` 选项的 `index`

Comment (ryuhangyeong) [reply packages/components/select/src/useSelect.ts:617]: ## Suggest

You can use the `findLastIndex` method more simply.

Comment (MrWeilian) [reply packages/components/select/src/useSelect.ts:617]: Are you mean the name of method ?

Comment (ryuhangyeong) [reply packages/components/select/src/useSelect.ts:617]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findLastIndex

Comment (MrWeilian) [reply packages/components/select/src/useSelect.ts:609]: hello, is some error in `test` when I was used `findLastIndex`, and I mock a simple `findLastIndex` polyfill to the test file. last, thank you for your suggestion again!😄

File: packages/components/select/src/useSelect.ts [hunk 2]
```diff
@@ -603,10 +604,15 @@ export const useSelect = (props, states: States, ctx) => {
     }
   }
 
+  const getLastNotDisabledIndex = (value) =>
+    value.findLastIndex((it) => !states.disabledOptions.has(it))
```
Comment (ryuhangyeong) [inline packages/components/select/src/useSelect.ts:608]: ## Suggest

It seems better to use lodash's findLastIndex than bypassing it with a mock .

If you accept this, please delete all test code changes.

```ts
import {
  findLastIndex,
} from 'lodash-unified'
```

Comment (ryuhangyeong) [reply packages/components/select/src/useSelect.ts:608]: Oh wait! I checked and the test still seems to be broken.

Comment (MrWeilian) [reply packages/components/select/src/useSelect.ts]: hello~ test code have been add.

**Review by ryuhangyeong — APPROVED** [review]

**Comment by ryuhangyeong** [comment]
LGTM 😃

**Comment by btea** [comment]
@MrWeilian Could you resolve the code conflict?

**Comment by btea** [comment]
@MrWeilian Don't worry, the test case is sometimes unstable, it will be fine after re-running.

**Review by btea — APPROVED** [review]

**Review by tolking — APPROVED** [review]

**Comment by wforguo** [comment]
七月份提的pr，没给我合，合了一个八月份的，妙啊

https://github.com/element-plus/element-plus/pull/13690

**Comment by MrWeilian** [comment]
- - 建议提 pr 的时候可以关联一下issue，并看看有没相关的解决。。。 #11995  是三月份提的。。。

**Pull request merged**